src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = new Book();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_SQL)) {
+            statement.setObject(1, id, Types.BIGINT);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = collectBook(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find book with id = " + id, e);
+        }
+        return Optional.of(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_SQL)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                bookList.add(collectBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find all books from DB " + e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BY_ID_SQL)) {
+            statement.setObject(1, book.getTitle(), Types.VARCHAR);
+            statement.setObject(2, book.getPrice(), Types.DECIMAL);
+            statement.setObject(3, book.getId(), Types.BIGINT);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("At least one parameter must be update,"
+                        + "but anyone was`nt updated");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update data in DB " + e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_BY_INDEX_SQL)) {
+            statement.setObject(1, id, Types.BIGINT);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete book with id = " + id, e);
+        }
+    }
+
+    private Book collectBook(ResultSet resultSet) {
+        Book book = new Book();
+        try {
+            book.setId(resultSet.getLong("id"));
+            book.setTitle(resultSet.getString("title"));
+            book.setPrice(resultSet.getBigDecimal("price"));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t collect data from DB");
```

Comment:
Consider passing the exception `e` as the second argument to the `DataProcessingException` constructor to preserve the stack trace.